### ISSUE TITLE
`vcpkg create` github template

### ISSUE THREAD

**Issue by vpaladino778** [microsoft/vcpkg#8509]
****

**Is your feature request related to a problem? Please describe.**
Right now, the `vcpkg create` command only creates a template if your using a direct .zip download link. There exists no template for a github portfile. 

**Proposed solution**
When the `vcpkg create` command is called, it will check if its a github.com url. If it is, instead of using the `scripts/template/portfile.in.cmake` template, it will use a new file, `scripts/template/portfile_github.in.cmake` which will contain a simple github portfile solution.

**Additional context**
I posted this issue because I want to implement this myself. I wanted to get the input/suggestions of the Microsoft employees that maintain the repository before I go start working on it. Let me know if this is something worth working on.

**Comment by PhoebeHui** [comment]
For me, It's good to extend current template, however, I'm not sure if it's necessary to create a separate portfile_github.in.cmake, since they have many same functions such as vcpkg_configure_cmake, vcpkg_install_cmake, vcpkg_fixup_cmake_targets, and handling copyright, and other common variables.

**Comment by vpaladino778** [comment]
@PhoebeHui Thanks for your reply! I figured a github specific one would be better so we could use `vcpkg_from_github` instead of `vcpkg_download_distfile` (Instead of making the user manually delete the unnecessary one out).  If we made a few adjustments to `vcpkg create` for github, it could be create a nearly complete portfile for simple github packages. Maybe it's a nonissue, just something i was thinking about.
